etc/kayobe/pulp.yml

```diff
@@ -134,69 +120,74 @@ stackhpc_pulp_publication_deb_development: "{{ stackhpc_pulp_distribution_deb_de
+  {%- set deb_repo = deb_repo | combine({"policy": repo.policy}) -%}
+  {%- endif -%}
+  {%- if "mirror" in repo -%}
+  {%- set deb_repo = deb_repo | combine({"mirror": repo.mirror}) -%}
+  {%- endif -%}
+  {%- if "distributions" in repo -%}
+  {%- set deb_repo = deb_repo | combine({"distributions": repo.distributions}) -%}
+  {%- endif -%}
+  {%- if "components" in repo -%}
+  {%- set deb_repo = deb_repo | combine({"components": repo.components}) -%}
+  {%- endif -%}
+  {%- if "architectures" in repo -%}
+  {%- set deb_repo = deb_repo | combine({"architectures": repo.architectures}) -%}
+  {%- endif -%}
+  {%- set _ = deb_repos.append(stackhpc_pulp_deb_repos_default_config | combine(deb_repo)) -%}
+  {%- endfor -%}
+  {{ deb_repos }}
 
-  # Ubuntu Cloud Archive (UCA) repositories
-  - name: "ubuntu-cloud-archive-development"
-    repository: Ubuntu Cloud Archive
-    base_path: "ubuntu-cloud-archive/development"
-    mode: verbatim
-    state: present
-    required: "{{ stackhpc_pulp_sync_ubuntu_focal | bool }}"
 
-  # Third-party repositories
-  - name: "docker-ce-for-ubuntu-development"
-    repository: Docker CE for Ubuntu
-    base_path: "docker-ce/ubuntu/development"
-    mode: verbatim
-    state: present
-    required: "{{ stackhpc_pulp_sync_ubuntu_focal | bool }}"
-
-# Development gets promoted to production.
-stackhpc_pulp_distribution_deb_production:
-  # Base Ubuntu Focal repositories
-  - name: "ubuntu-focal-production"
-    repository: Ubuntu focal
-    base_path: "ubuntu/focal/production"
-    mode: verbatim
-    state: present
-    required: "{{ stackhpc_pulp_sync_ubuntu_focal | bool }}"
-  - name: "ubuntu-focal-security-production"
-    repository: Ubuntu focal security
-    base_path: "ubuntu/focal-security/production"
-    mode: verbatim
-    state: present
-    required: "{{ stackhpc_pulp_sync_ubuntu_focal | bool }}"
+stackhpc_pulp_deb_dists_default_config:
+  state: present
+  mode: verbatim
 
-  # Ubuntu Cloud Archive (UCA) repositories
-  - name: "ubuntu-cloud-archive-production"
-    repository: Ubuntu Cloud Archive
-    base_path: "ubuntu-cloud-archive/production"
-    mode: verbatim
-    state: present
-    required: "{{ stackhpc_pulp_sync_ubuntu_focal | bool }}"
+# Development tracks the release repositories.
+stackhpc_pulp_distribution_deb_development: >-
+  {%- set dev_dists = [] -%}
+  {%- for repo in stackhpc_pulp_deb_repos -%}
+  {%- set name = repo.distribution_name ~ 'development' -%}
+  {%- set base_path = repo.base_path ~ 'development' -%}
+  {%- set dev_dist = {
+                      "name": name,
+                      "repository": repo.name,
+                      "base_path": base_path,
+                      "required": repo.required
+                      }
+                    -%}
+  {%- if "mode" in repo -%}
+  {%- set dev_dist = dev_dist | combine({"mode": repo.mode}) -%}
+  {%- endif -%}
+  {%- set _ = dev_dists.append(stackhpc_pulp_deb_repos_default_config | combine(dev_dist)) -%}
```

Comment:
```suggestion
  {%- set _ = dev_dists.append(stackhpc_pulp_deb_dists_default_config | combine(dev_dist)) -%}
```

Reply:
Missed this when I split out the default repo config and default dists config :facepalm:. Updated.